pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -394,28 +345,3 @@ func (uc *UpgradeCluster) addVMToAgentPool(vm compute.VirtualMachine, isUpgradab
 
 	return nil
 }
-
```

Comment:
I actually kinda think we should keep this in, uncomment it, and standardize to persisting the template that we deploy to upgrade always.

But perhaps for a later exercise.

Reply:
We already have logic for saving the modified template in upgrade.go https://github.com/CecileRobertMichon/aks-engine/blob/0b7a9702300ed5d033874a6ced2d2a140867fe18/cmd/upgrade.go#L281 so this commented out code is redundant.

I think we need to improve the logic for persisting templates and instead of passing them to a file, saving them in the RG or something like that but, like you said, for a later exercise.